internal/pkg/utils/orderedmap/orderedmap_test.go

```diff
@@ -558,3 +564,254 @@ func TestOrderedMap_empty_map(t *testing.T) {
+	assert.NoError(t, root.SetNested(`str`, `value`))
+	err := root.SetNested(`str.key`, `value`)
+	assert.Error(t, err)
+	assert.Equal(t, `key "str": expected object found "string"`, err.Error())
+	err = root.SetNestedPath(Key{MapStep(`str`), MapStep(`key`)}, `value`)
+	assert.Error(t, err)
+	assert.Equal(t, `key "str": expected object found "string"`, err.Error())
+
+	expected := `
+{
+  "foo1": "bar1",
+  "foo2": "bar2",
+  "nested": {
+    "foo3": "bar3",
+    "foo4": "bar4",
+    "missing": {
+      "key": "value"
+    }
+  },
+  "str": "value"
+}
+`
+	assert.Equal(t, strings.TrimLeft(expected, "\n"), jsonutils.MustEncodeString(root, true))
+}
+
+func TestOrderedMap_VisitAllRecursive(t *testing.T) {
+	t.Parallel()
+	input := `
+{
+    "foo1": "bar1",
+    "foo2": "bar2",
+    "nested1": {
+        "foo3": "bar3",
+        "foo4": "bar4",
+        "nested2": {
+            "key": "value"
+        },
+        "slice": [
+            123,
+            "abc",
+            {
+                "nested3": {
+                    "foo5": "bar5"
+                }
+            },
+            {
+                "subSlice": [
+                    456,
+                    "def",
+                    {
+                        "nested4": {
+                            "foo6": "bar6"
+                        }
+                    }
+                ]
+            }
+        ]
+    },
+    "str": "value"
+}
+`
+
+	expected := `
+path=foo1, parent=*orderedmap.OrderedMap, value=string
+path=foo2, parent=*orderedmap.OrderedMap, value=string
+path=nested1, parent=*orderedmap.OrderedMap, value=*orderedmap.OrderedMap
+path=nested1.foo3, parent=*orderedmap.OrderedMap, value=string
+path=nested1.foo4, parent=*orderedmap.OrderedMap, value=string
+path=nested1.nested2, parent=*orderedmap.OrderedMap, value=*orderedmap.OrderedMap
+path=nested1.nested2.key, parent=*orderedmap.OrderedMap, value=string
+path=nested1.slice, parent=*orderedmap.OrderedMap, value=[]interface {}
+path=nested1.slice[0], parent=[]interface {}, value=float64
+path=nested1.slice[1], parent=[]interface {}, value=string
+path=nested1.slice[2], parent=[]interface {}, value=*orderedmap.OrderedMap
+path=nested1.slice[2].nested3, parent=*orderedmap.OrderedMap, value=*orderedmap.OrderedMap
+path=nested1.slice[2].nested3.foo5, parent=*orderedmap.OrderedMap, value=string
+path=nested1.slice[3], parent=[]interface {}, value=*orderedmap.OrderedMap
+path=nested1.slice[3].subSlice, parent=*orderedmap.OrderedMap, value=[]interface {}
+path=nested1.slice[3].subSlice[0], parent=[]interface {}, value=float64
+path=nested1.slice[3].subSlice[1], parent=[]interface {}, value=string
+path=nested1.slice[3].subSlice[2], parent=[]interface {}, value=*orderedmap.OrderedMap
+path=nested1.slice[3].subSlice[2].nested4, parent=*orderedmap.OrderedMap, value=*orderedmap.OrderedMap
+path=nested1.slice[3].subSlice[2].nested4.foo6, parent=*orderedmap.OrderedMap, value=string
+path=str, parent=*orderedmap.OrderedMap, value=string
```

Comment:
Metoda `VisitAllRecursive` zavola callback pre kazdu najdenu polozku, pouzije sa to aj pri generovani sablony.
Aktualne sa pouziva pre `encrypt` - hladanie nezasifrovanej hodnoty.